From 2b3a50ec84bf13853352c17c816a2be0ab47adc8 Mon Sep 17 00:00:00 2001
From: Constantin Musca <constantin.musca@intel.com>
Date: Thu, 22 Jan 2015 15:43:44 +0200
Subject: [PATCH 1/1] gt9xx: reduce sleep time between STATUS read retries

Sometimes a tap is detected as touch-and-hold because we
we sleep 1.0-1.5ms between STATUS read retries. There are cases
(not often) when the STATUS register is not valid and we lose an
BTN_TOUCH UP event while sleeping.
Reduce the sleep time to (0.5-1.0)ms.

Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3232
Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3178
Change-Id: I30526abfa8da4085fb0c4044c43ddb8e16ee5660
Signed-off-by: Constantin Musca <constantin.musca@intel.com>
---
 drivers/input/touchscreen/gt9xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/gt9xx.c b/drivers/input/touchscreen/gt9xx.c
index 3f990f1..4f7738c 100644
--- a/drivers/input/touchscreen/gt9xx.c
+++ b/drivers/input/touchscreen/gt9xx.c
@@ -233,7 +233,7 @@ static irqreturn_t gt9xx_thread_handler(int irq, void *arg)
 		 * is still pressing screen and we wait a bit. So we'll wait
 		 * and retry few times before discard the buffer for good.
 		 */
-		usleep_range(1000, 1500);
+		usleep_range(500, 1000);
 	}
 
 	if (!retries) {
-- 
1.9.1

